Source/API/Extras/CBLIncrementalStore.m

```diff
@@ -697,89 +839,67 @@ - (NSManagedObjectID *)newObjectIDForEntity:(NSEntityDescription *)entity refere
-        
+
         for (NSPropertyDescription *property in properties) {
-            
+
             if ([property isKindOfClass:[NSRelationshipDescription class]]) {
                 NSRelationshipDescription *rel = (NSRelationshipDescription*)property;
-                
                 if (rel.isToMany) {
-                    
-                    NSMutableArray *entityNames = nil;
-                    if (rel.destinationEntity.subentities.count > 0) {
-                        entityNames = [NSMutableArray arrayWithCapacity:3];
-                        for (NSEntityDescription *subentity in rel.destinationEntity.subentities) {
-                            [entityNames addObject:subentity.name];
-                        }
+                    NSMutableArray *entityNames = [NSMutableArray arrayWithObject:rel.destinationEntity.name];
+                    for (NSEntityDescription *subentity in rel.destinationEntity.subentities) {
+                        [entityNames addObject:subentity.name];
                     }
-                    
+
                     NSString *viewName = CBLISToManyViewNameForRelationship(rel);
-                    NSString *destEntityName = rel.destinationEntity.name;
-                    NSString *inverseRelNameLower = rel.inverseRelationship.name;
-                    if (entityNames.count == 0) {
-                        CBLView *view = [self.database viewNamed:viewName];
-                        [view setMapBlock:^(NSDictionary *doc, CBLMapEmitBlock emit) {
-                            if ([[doc objectForKey:kCBLISTypeKey] isEqual:destEntityName] && [doc objectForKey:inverseRelNameLower]) {
-                                emit([doc objectForKey:inverseRelNameLower], nil);
-                            }
-                        }
-                                  version:@"1.0"];
-                        
-                        [self _setViewName:viewName forFetchingProperty:inverseRelNameLower fromEntity:destEntityName];
-                        
-                    } else {
-                        CBLView *view = [self.database viewNamed:viewName];
-                        [view setMapBlock:^(NSDictionary *doc, CBLMapEmitBlock emit) {
-                            if ([entityNames containsObject:[doc objectForKey:kCBLISTypeKey]] && [doc objectForKey:inverseRelNameLower]) {
-                                emit([doc objectForKey:inverseRelNameLower], nil);
-                            }
+                    NSRelationshipDescription *invRel = rel.inverseRelationship;
+                    NSString *invertRelPropName = invRel ?
+                    invRel.name : CBLISInverseRelationshipKeyForRelationName(rel.name);
+                    CBLView *view = [self.database viewNamed:viewName];
+                    [view setMapBlock:^(NSDictionary *doc, CBLMapEmitBlock emit) {
+                        if ([entityNames containsObject:[doc objectForKey:kCBLISTypeKey]] &&
```

Comment:
This will throw an exception if the document doesn't contain kCBLISTypeKey. (Looks like the previous revision had the same problem, so this isn't a regression.)